metricbeat/module/kafka/partition/partition.go

```diff
@@ -37,57 +36,60 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch partition stats list from kafka
 func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 
-	if m.client == nil {
+	if m.broker == nil {
 		config := sarama.NewConfig()
 		config.Net.DialTimeout = m.Module().Config().Timeout
 		config.Net.ReadTimeout = m.Module().Config().Timeout
 		config.ClientID = "metricbeat"
 
-		client, err := sarama.NewClient([]string{m.Host()}, config)
+		broker := sarama.NewBroker(m.Host())
+		err := broker.Open(config)
 		if err != nil {
 			return nil, err
 		}
-		m.client = client
+		m.broker = broker
 	}
 
-	topics, err := m.client.Topics()
+	response, err := m.broker.GetMetadata(&sarama.MetadataRequest{})
 	if err != nil {
 		return nil, err
 	}
 
 	events := []common.MapStr{}
-	for _, topic := range topics {
-		partitions, err := m.client.Partitions(topic)
-		if err != nil {
-			logp.Err("Fetch partition info for topic %s: %s", topic, err)
-		}
+	for _, topic := range response.Topics {
 
-		for _, partition := range partitions {
-			newestOffset, err := m.client.GetOffset(topic, partition, sarama.OffsetNewest)
-			if err != nil {
-				logp.Err("Fetching newest offset information for partition %s in topic %s: %s", partition, topic, err)
-			}
+		for _, partition := range topic.Partitions {
 
-			oldestOffset, err := m.client.GetOffset(topic, partition, sarama.OffsetOldest)
-			if err != nil {
-				logp.Err("Fetching oldest offset information for partition %s in topic %s: %s", partition, topic, err)
-			}
+			offsetRequest := &sarama.OffsetRequest{}
+			// Get 2 offsets, assumption is newest first in array, oldest second
+			offsetRequest.AddBlock(topic.Name, partition.ID, sarama.OffsetNewest, 2)
 
-			broker, err := m.client.Leader(topic, partition)
-			if err != nil {
-				logp.Err("Fetching brocker for partition %s in topic %s: %s", partition, topic, err)
-			}
+			offsetResponse, _ := m.broker.GetAvailableOffsets(offsetRequest)
```

Comment:
Does GetAvailableOffsets add the current broker ID to the API request?

Checking the ['encode' method](https://github.com/Shopify/sarama/blob/master/offset_request.go#L29), it seems ReplicaID is always -1. Does it mean, we query offsets from leader?